### DISCUSSION TITLE
Matching Attachment Name on 2 Criteria

### DISCUSSION THREAD

**Discussion by grafpaper1987** [ahochsteger/gmail-processor#351] [Q&A] [answered]
Hi there! This is an incredibly helpful tool, so thank you!

I'm trying (but struggling) to limit the attachments that get saved based on two criteria.

For example, an email contains attachments with the following 3 names:

**- 003101172267-FE-00100001010001679522.xml**
- 003101172267-MH-00100001010001679522.xml
- 003101172267-FE-00100001010001679522.PDF

I'm trying to only save down the bolded one. I can't seem to figure out how to tell it only to save files:

1. Containing "FE" (anywhere in the file name) and 
2. Ending in ".xml"

The following config (understandably) picks up both XML files:

"attachments": [
              {
                "match": {
                  "name": "xml"
                },

What is the correct modification to this section of the config to limit the matches based on my two bulleted criteria above?

It feels like I'm probably missing something simple here. Any guidance is appreciated. Thank you.

**Comment by ahochsteger** [comment, marked as answer]
@grafpaper1987 nice to see that it's helpful for you! :smile:

Your use case can easily be solved using [Regular Expressions](https://en.wikipedia.org/wiki/Regular_expression).
This is the simplest regex that matches the whole filename you want to match: `^.*-FE-.*\.xml$`
I created [this example on regex101.com](https://regex101.com/r/SkuqcK/4) so you can play with the regex and different example file names and see which will match and which not.
It also explains the syntax and matching parts of the regular expression.

In case you want to do more with the numbers and characters that are part of the filename you can use Named Capturing Groups that are
able to extract certain parts of the name and make them available for further processing steps (e.g. as part of the drive path or filename).
The is an advanced version would look like this: `^(?<prefixNr>.*)-(?<type>FE)-(?<suffixNr>.*)\.(?<ext>xml)$`
Again you can use [this example on regex101.com](https://regex101.com/r/QkQe7s/2) to play around with it.

I hope this helps.

**Comment by grafpaper1987** [comment]
Thank you so much! The regex101.com example is incredibly helpful.
